NormalMsgDelayIT is flaky

There was 1 failure:
1) testDelayLevel3(org.apache.rocketmq.test.delay.NormalMsgDelayIT)
java.lang.AssertionError: Timer is not correct expected:<true> but was:<false>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at org.apache.rocketmq.test.delay.NormalMsgDelayIT.testDelayLevel3(NormalMsgDelayIT.java:96)

See full CI log: https://github.com/apache/rocketmq/actions/runs/3235189115/jobs/5299272721

--- a/test/src/main/java/org/apache/rocketmq/test/util/VerifyUtils.java
+++ b/test/src/main/java/org/apache/rocketmq/test/util/VerifyUtils.java
@@ -82,22 +82,17 @@ public class VerifyUtils {
         return verifyBalance(msgSize, 0.1f, recvSize);
     }
 
-    public static boolean verifyDelay(long delayTimeMills, Collection<Object> recvMsgTimes,
-        int errorMills) {
+    public static boolean verifyDelay(long delayTimeMills, long nextLevelDelayTimeMills,
+        Collection<Object> recvMsgTimes) {
         boolean delay = true;
         for (Object timeObj : recvMsgTimes) {
             long time = (Long) timeObj;
-            if (Math.abs(time - delayTimeMills) > errorMills) {
+            if (time < delayTimeMills || time > nextLevelDelayTimeMills) {
                 delay = false;
                 logger.info(String.format("delay error:%s", Math.abs(time - delayTimeMills)));
             }
         }
         return delay;
-    }
-
-    public static boolean verifyDelay(long delayTimeMills, Collection<Object> recvMsgTimes) {
-        int errorMills = 500;
-        return verifyDelay(delayTimeMills, recvMsgTimes, errorMills);
     }
 
     public static boolean verifyOrder(Collection<Collection<Object>> queueMsgs) {

--- a/test/src/test/java/org/apache/rocketmq/test/delay/NormalMsgDelayIT.java
+++ b/test/src/test/java/org/apache/rocketmq/test/delay/NormalMsgDelayIT.java
@@ -17,6 +17,7 @@
 
 package org.apache.rocketmq.test.delay;
 
+import java.util.List;
 import org.apache.rocketmq.logging.org.slf4j.Logger;
 import org.apache.rocketmq.logging.org.slf4j.LoggerFactory;
 import org.apache.rocketmq.test.client.rmq.RMQNormalConsumer;
@@ -28,8 +29,6 @@ import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
-
-import java.util.List;
 
 public class NormalMsgDelayIT extends DelayConf {
     private static Logger logger = LoggerFactory.getLogger(NormalMsgDelayIT.class);
@@ -63,7 +62,7 @@ public class NormalMsgDelayIT extends DelayConf {
         Assert.assertEquals("Not all are consumed", 0, VerifyUtils.verify(producer.getAllMsgBody(),
             consumer.getListener().getAllMsgBody()));
         Assert.assertEquals("Timer is not correct", true,
-            VerifyUtils.verifyDelay(DELAY_LEVEL[delayLevel - 1] * 1000,
+            VerifyUtils.verifyDelay(DELAY_LEVEL[delayLevel - 1] * 1000, DELAY_LEVEL[delayLevel] * 1000,
                 ((RMQDelayListener) consumer.getListener()).getMsgDelayTimes()));
     }
 
@@ -79,7 +78,7 @@ public class NormalMsgDelayIT extends DelayConf {
         Assert.assertEquals("Not all are consumed", 0, VerifyUtils.verify(producer.getAllMsgBody(),
             consumer.getListener().getAllMsgBody()));
         Assert.assertEquals("Timer is not correct", true,
-            VerifyUtils.verifyDelay(DELAY_LEVEL[delayLevel - 1] * 1000,
+            VerifyUtils.verifyDelay(DELAY_LEVEL[delayLevel - 1] * 1000, DELAY_LEVEL[delayLevel] * 1000,
                 ((RMQDelayListener) consumer.getListener()).getMsgDelayTimes()));
     }
 
@@ -95,7 +94,7 @@ public class NormalMsgDelayIT extends DelayConf {
         Assert.assertEquals("Not all are consumed", 0, VerifyUtils.verify(producer.getAllMsgBody(),
             consumer.getListener().getAllMsgBody()));
         Assert.assertEquals("Timer is not correct", true,
-            VerifyUtils.verifyDelay(DELAY_LEVEL[delayLevel - 1] * 1000,
+            VerifyUtils.verifyDelay(DELAY_LEVEL[delayLevel - 1] * 1000, DELAY_LEVEL[delayLevel] * 1000,
                 ((RMQDelayListener) consumer.getListener()).getMsgDelayTimes()));
     }
 
@@ -111,7 +110,7 @@ public class NormalMsgDelayIT extends DelayConf {
         Assert.assertEquals("Not all are consumed", 0, VerifyUtils.verify(producer.getAllMsgBody(),
             consumer.getListener().getAllMsgBody()));
         Assert.assertEquals("Timer is not correct", true,
-            VerifyUtils.verifyDelay(DELAY_LEVEL[delayLevel - 1] * 1000,
+            VerifyUtils.verifyDelay(DELAY_LEVEL[delayLevel - 1] * 1000, DELAY_LEVEL[delayLevel] * 1000,
                 ((RMQDelayListener) consumer.getListener()).getMsgDelayTimes()));
     }
 }

